### PR TITLE
Raise error if disjoint time co-ordinates inside _align_obs_footprint

### LINKED ISSUE #882: `_align_obs_footprint` gives confusing error if the obs and footprint have disjoint time coordinates
### What is your issue?

If the obs and footprint added to a `ModelScenario` have non-overlapping date ranges, then an error is thrown when resampling because we slice the obs and footprint to a common date range (modulo some time period adjustments):
```
obs_data = obs_data.sel(time=slice(start_obs_slice, end_slice))
footprint_data = footprint_data.sel(time=slice(start_footprint_slice, end_slice))
```
It would be useful to check that after this step, `obs_data` and `footprint_data` are not empty.

### PR DESCRIPTION
**Pull request by SutarPrasad**
* **Summary of changes** (Bug fix, feature, docs update, ...)

* **Please check if the PR fulfills these requirements**

- [x] Closes #882 (Replace xxxx with the Github issue number)
- [x] [Tests added and passed](https://docs.openghg.org/development/python_devel.html#testing) if fixing a bug or adding a new feature
- [x] All code checks passing - `black --line-length 110` run over code, `mypy` and `flake8` not showing any errors
- [x] Documentation and tutorials updated/added
- [x] Added an entry in the latest `CHANGELOG.md` file if fixing a bug or adding a new feature
- [x] Added any new requirements to `requirements.txt` and `recipes/meta.yaml`

### CODE REVIEW

File: openghg/analyse/_scenario.py [hunk 1]
```diff
@@ -712,6 +712,8 @@ def _align_obs_footprint(self, resample_to: str = "coarsest", platform: Optional
         obs_data = obs_data.sel(time=slice(start_obs_slice, end_slice))
         footprint_data = footprint_data.sel(time=slice(start_footprint_slice, end_slice))
 
+        if obs_data is None or footprint_data is None:
```
Comment (gareth-j) [inline openghg/analyse/_scenario.py:715]: Here we don't get a `None` if the datasets don't contain the requried dates. We get a `Dataset` with a time dimension of size zero. We could check for `obs_data.time.size == 0` or something?

**Review by gareth-j — CHANGES_REQUESTED** [review]
Need to update check of date overlap.

File: openghg/analyse/_scenario.py (hunk 1 above)
Comment (SutarPrasad) [reply openghg/analyse/_scenario.py]: Thanks for this suggestion, I struggled to add a test because there was never None. Updated as per your suggestion.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased](https://github.com/openghg/openghg/compare/0.8.0...HEAD)
 
+### Fixed
+
+- Added check to verify if `obs` and `footprint` have disjoint time co-ordinates then raise ValueError - [PR #954](https://github.com/openghg/openghg/pull/954)
```
Comment (gareth-j) [inline CHANGELOG.md:12]:
```suggestion
- Added check to verify if `obs` and `footprint` have overlapping time coordinates when creating a `ModelScenario` object, if not then raise `ValueError` - [PR #954](https://github.com/openghg/openghg/pull/954)
```

File: openghg/analyse/_scenario.py [hunk 3]
```diff
@@ -724,6 +724,8 @@ def _align_obs_footprint(self, resample_to: str = "coarsest", platform: Optional
         obs_data = obs_data.sel(time=slice(start_obs_slice, end_slice))
         footprint_data = footprint_data.sel(time=slice(start_footprint_slice, end_slice))
 
+        if obs_data.time.size == 0 or footprint_data.time.size == 0:
+            raise ValueError("Obs Data and Footprint Data don't overlap")
```
Comment (gareth-j) [inline openghg/analyse/_scenario.py:728]:
```suggestion
            raise ValueError("Obs data and Footprint data don't overlap")
```

File: tests/analyse/test_scenario.py [hunk 4]
```diff
@@ -846,6 +847,62 @@ def test_model_modelled_obs_ch4(model_scenario_ch4_dummy, footprint_dummy, flux_
     modelled_mf = combined_dataset["mf_mod"].values
     assert np.allclose(modelled_mf, expected_modelled_mf)
 
+@pytest.fixture
+def obs_dummy():
```
Comment (gareth-j) [inline tests/analyse/test_scenario.py:851]: I don't think this needs to be a fixture if it's only used once.

File: tests/analyse/test_scenario.py [hunk 5]
```diff
@@ -846,6 +847,62 @@ def test_model_modelled_obs_ch4(model_scenario_ch4_dummy, footprint_dummy, flux_
...
+    """
+    Create example ObsData object with dummy data
+     - Species is methane (ch4)
+     - Hourly frequency for 2012-01-01 - 2012-01-02 (48 time points)
+     - "mf" values are from 1, 48
+    """
+    from openghg.dataobjects import ObsData
```
Comment (gareth-j) [inline tests/analyse/test_scenario.py:858]: Move this import to the start of the file

File: tests/analyse/test_scenario.py [hunk 6]
```diff
@@ -846,6 +847,62 @@ def test_model_modelled_obs_ch4(model_scenario_ch4_dummy, footprint_dummy, flux_
...
+    # - data_type
+    metadata = attributes
+    metadata["object_store"] = "/tmp/test-store-123"
+
+    obsdata = ObsData(data=data, metadata=metadata)
+
+    return obsdata
+
+@pytest.fixture
+def model_scenario_dummy(obs_dummy, footprint_dummy, flux_ch4_dummy, bc_ch4_dummy):
```
Comment (gareth-j) [inline tests/analyse/test_scenario.py:891]: This doesn't need to be a fixture and can just be created in the test.

**Review by gareth-j — CHANGES_REQUESTED** [review]
Just a few changes to the tests and some text.

File: tests/analyse/test_scenario.py (hunk 5 above)
Comment (SutarPrasad) [reply tests/analyse/test_scenario.py]: done

File: tests/analyse/test_scenario.py (hunk 6 above)
Comment (SutarPrasad) [reply tests/analyse/test_scenario.py]: done

File: tests/analyse/test_scenario.py (hunk 4 above)
Comment (SutarPrasad) [reply tests/analyse/test_scenario.py]: done

**Review by gareth-j — APPROVED** [review]
Looks good!
